### PR TITLE
[My Team] Allow senator to delegate their access

### PR DESCRIPTION
**Pull request by kevin-verschaeve**

### CODE REVIEW

File: src/Repository/GeoFilterTrait.php [hunk 1]
```diff
@@ -26,6 +26,15 @@ private function applyDeputyGeoFilter(QueryBuilder $qb, Adherent $deputy, string
         $this->applyGeoFilter($qb, [$deputy->getManagedDistrict()->getReferentTag()], $alias);
     }
 
+    private function applySenatorGeoFilter(QueryBuilder $qb, Adherent $senator, string $alias): void
+    {
+        if (!$senator->isDeputy()) {
```
Comment (ottaviano) [inline src/Repository/GeoFilterTrait.php]: `isSenator`? 
it seems that `applySenatorGeoFilter` is unused

File: templates/senator/_layout.html.twig [hunk 2]
```diff
@@ -27,18 +27,43 @@
                         </h1>
                     </div>
                     <nav class="manager-header__menu">
+                        {% set isSenator = is_granted('ROLE_SENATOR') %}
```
Comment (ottaviano) [inline templates/senator/_layout.html.twig:30]: I think we could avoid this check as we already are in `senator/_layout.html.twig`.
what about doing something like this: 

```twig
<ul>
  {% if is_granted('HAS_DELEGATED_ACCESS_...', 'senator') %}
    <li delegated_route>
  {% else %}
    <li senator_route>
  {% endif %}

  ...

</ul>
```

File: src/Repository/GeoFilterTrait.php [hunk 3]
```diff
@@ -26,6 +26,15 @@ private function applyDeputyGeoFilter(QueryBuilder $qb, Adherent $deputy, string
         $this->applyGeoFilter($qb, [$deputy->getManagedDistrict()->getReferentTag()], $alias);
     }
 
+    private function applySenatorGeoFilter(QueryBuilder $qb, Adherent $senator, string $alias): void
```
Comment (ottaviano) [inline src/Repository/GeoFilterTrait.php:29]: it seems to be unused

File: templates/senator/_layout.html.twig (hunk 2 above)
Comment (kevin-verschaeve) [reply templates/senator/_layout.html.twig:30]: ok done. I have still let it on the last tab, as "my team" should be displayed only to senator

File: src/Repository/GeoFilterTrait.php (hunk 3 above)
Comment (kevin-verschaeve) [reply src/Repository/GeoFilterTrait.php:29]: wow,  good catch ! I indeed forgot to use it ! Fixed

**Pull request merged**